app/controllers/users/mfa_selection_controller.rb

```diff
@@ -4,7 +4,10 @@ class MfaSelectionController < ApplicationController
     include MfaSetupConcern
 
     before_action :authenticate_user
-    before_action :confirm_user_authenticated_for_2fa_setup
+    before_action :confirm_user_authenticated_for_2fa_setup, if: proc {
+                                                                   !params[:non_restricted].
```

Comment:
I think there may be a small typo or error here.

If `params[:non_restricted]` is `nil`, the expression `!params[:non_restricted].to_s` evaluates to the boolean `false`, which is then compared against the string 'true'.

This can result in an infinite redirect when testing under the following conditions when `IdentityConfig.store.kantara_2fa_phone_restricted` is `true` and the user only has sms as enabled.

Reply:
To be more clear, when `params[:non_restricted]` is nil, the following expression 
`!params[:non_restricted].to_s != 'true'` evaluates to `true`, triggering a call to `confirm_user_authenticated_for_2fa_setup`. Around here an infinite loop is reached.

Maybe I should have `IdentityConfig.store.select_multiple_mfa_options` enabled?

Reply:
There's a loop but maybe I spoke too soon. Appears to be related to the following before_action. I haven't transcribed the full chain of calls. Also uncertain if I have the correct flags set.

Reply:
Good catch. The flag configuration to trigger this loop is:
1. select_multiple_mfa_options: 'false'
2. kantara_2fa_phone_restricted: 'true'

I don't think we'd ever have kantara be true with multiple mfa set to false so we'd never trigger this loop. @zachmargolis this approach was your suggestion, do you think we should go further down this path or is this complexity enough of a reason to revert back to the separate method? Or do you think that because that combination of feature flags should never happen, that we shouldn't worry about this?

Reply:
I guess if we tied ourselves in a knot like this, maybe a separate method makes sense. Thanks for humoring me and giving it a shot.

If we do two separate actions, but maybe just have an entirely separate template? Since it seems like the presenter is basically switching between two completely separate modes?

Reply:
I think the presenter with a single template is still a good approach here since the views are identical, other than the content, which is different depending on if this is on initial account setup or a returning user. This reusability will help us in the future as well, if/when we add a third set of content for suggesting, instead of enforcing, a second mfa for returning users.